### ISSUE TITLE
Event handlers have to be unregistered when a plugin or input stops

### ISSUE THREAD

**Issue by hlxid** [codeoverflow-org/chatoverflow#89]
### Description
All event handlers that are registered by plugins using some inputs aren't unregistered at all.
Even after a plugin has stopped the event handler is still called for all received events.

The event handlers by all event inputs that they register in their connectors also have to be unregistered at some point, probably in their `shutdown` method.

### How to reproduce
1. Have a simple plugin like this: 
```scala
import org.codeoverflow.chatoverflow.api.plugin.{PluginImpl, PluginManager}

class SimpleTwitchChatPlugin(manager: PluginManager) extends PluginImpl(manager) {
  private val in = require.input.twitchChat("twitchIn", "The twitch input, of which all messages will be displayed", false)
  private val channel = "skate702"

  override def setup(): Unit = {
    in.get().setChannel(channel)
    in.get().registerChatMessageReceiveEventHandler(ev => println(ev.getMessage))
  }

  override def loop(): Unit = ()
  override def shutdown(): Unit = ()
}
```
2. Start the plugin. Notice that every chat message is printed once, as it should.
3. Restart the plugin, but not the framework. Notice that every chat message is printed **four** times.

It's printed four times, because the input has registered its internal handler twice and the plugin has the handler registered in the input twice. Each internal handler is calling all handlers by the plugin, which results in four calls.

**Comment by J0B10** [comment]
Autsch. 🙈

**Comment by J0B10** [comment]
A tricky one that probably existed before the rework of event inputs.

I did rework the EventInputImpl to clear all handlers on shutdown (see commit ee0d583) which didn't help.

Then I took a look at the connectors and inputs implementations and reworked them all to unregister their listeners (see commit 4de98d7) but still each message is handled multiple times.

At least those changes decreased the amount of handlers from four to two, but this is still one to much😅🤦‍♂ .

I then did some debugging and I currently don't really understand whats going on there:

I used the testall plugin for testing with the discord service, as this is the easiest to use for me.
I did use a breakpoint at [line 185 of PluginInstance](https://github.com/codeoverflow-org/chatoverflow/blob/4de98d7d01eb8e4e9fd7ee41453a013bbf119fa6/src/main/scala/org/codeoverflow/chatoverflow/instance/PluginInstance.scala#L185) and stopped the plugin twice.  
On the second stop I got the following result for the variables:  

> ![grafik](https://user-images.githubusercontent.com/17405009/61173160-b9f10e80-a58f-11e9-911c-52f9f5a1059c.png)  

What caught my eye is that **the same input object** was registered **twice** as listener.  

I don't really get why this happens and currently have no idea how it could be fixed.
